core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -0,0 +1,490 @@
+package rollups
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/v2/common/client"
+	"github.com/smartcontractkit/chainlink/v2/common/config"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/assets"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+)
+
+// Reads L2-specific precompiles and caches the l1GasPrice set by the L2.
+type optimismL1Oracle struct {
+	services.StateMachine
+	client     l1OracleClient
+	pollPeriod time.Duration
+	logger     logger.SugaredLogger
+	chainType  config.ChainType
+
+	l1GasPriceAddress   string
+	gasPriceMethod      string
+	l1GasPriceMethodAbi abi.ABI
+	l1GasPriceMu        sync.RWMutex
+	l1GasPrice          priceEntry
+
+	l1GasCostAddress   string
```

Comment:
I think we can remove this field from this l1 oracle. This was introduced in the generic one to have a distinction between the contract we call for gas price vs gas cost because Arbitrum had separate contracts. Now that we have chain specific l1 oracles, we don't have to support that here for OP. We can also rename `l1GasPriceAddress` to something more generic like `l1OracleAddress` since all of the method are served out of the same contract. You'll notice we set the same address for both below in the builder anyways.

Reply:
Ok, removing for Scroll too